File: C#/817. Linked List Components.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public int NumComponents(ListNode head, int[] nums) {
        Dictionary<int, int> dict = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++){
            dict[nums[i]]=i;
        }
        int res = 0;
        while(head!=null){
            if(dict.ContainsKey(head.val)){
                while(head.next!=null && dict.ContainsKey(head.next.val)){
                    head = head.next;
                }
                res++;
            }
            head = head.next;
        }
        return res;
    }
}

File: JavaScript/817. Linked List Components.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number[]} nums
 * @return {number}
 */
var numComponents = function(head, nums) {
    var mp = new Map();
    for(var i of nums){
        mp[i]++;
    }
    var res = 0;
    while(head!=null){
        if(mp[head.val]!=null){
            while(head.next!=null && mp[head.next.val]!=null){
                head = head.next;
            }
            res++;
        }
        head = head.next;
    }
    return res;
};
